File: core/src/main/java/com/palomiklo/streampulse/connection/ConnectionHolder.java
package com.palomiklo.streampulse.connection;

import com.palomiklo.streampulse.config.ConfigLoader;
import com.palomiklo.streampulse.context.AsynchronousContext;
import com.palomiklo.streampulse.listener.Listener;

public record ConnectionHolder(Connection connection, AsynchronousContext ctx) {

    public ConnectionHolder  {
        ConfigLoader.initialize();
    }

    public static ConnectionHolder create(Connection connection, AsynchronousContext ctx) {
        ConnectionHolder chldr = new ConnectionHolder(connection, ctx);
        Listener.addListeners(ctx.actx(), chldr);
        return chldr;
    }
}

File: core/src/main/kotlin/com/palomiklo/streampulse/connection/ConnectionHolder.kt
package com.palomiklo.streampulse.connection

import com.palomiklo.streampulse.config.ConfigLoader.Companion.initializeConfig
import com.palomiklo.streampulse.context.AsynchronousContext
import com.palomiklo.streampulse.listener.Listener

@JvmRecord
data class ConnectionHolder(val connection: Connection, val ctx: AsynchronousContext) {
    init {
        initializeConfig()
    }

    companion object {
        fun create(connection: Connection, ctx: AsynchronousContext): ConnectionHolder =
            ConnectionHolder(connection, ctx).also { chldr -> Listener.addListeners(ctx.actx, chldr) }
    }
}
